org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/configuration/CompositeConfiguration.java

```diff
@@ -0,0 +1,33 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Gradle Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ******************************************************************************/
+
+package org.eclipse.buildship.core.internal.configuration;
+
+import java.io.File;
+
+import org.eclipse.core.runtime.IAdaptable;
+
+/**
+ * Configuration for a workspace composite in a Gradle build.
+ *
+ * @author Sebastian Kuzniarz
+ */
+public interface CompositeConfiguration {
+
+    File getCompositeDir();
+
+    IAdaptable[] getProjectList();
```

Comment:
⭕ Something is off here. I think we should just store the root location of the included builds. The return type of this method should be `List<File>`.

Reply:
I guess that'd make including external projects a lot easier!